modules/core/src/directx.cpp

```diff
@@ -1561,26 +1659,23 @@ void convertFromD3D10Texture2D(ID3D10Texture2D* pD3D10Texture2D, OutputArray dst
 }
 
 #if defined(HAVE_DIRECTX) && defined(HAVE_OPENCL)
-clCreateFromDX9MediaSurfaceKHR_fn clCreateFromDX9MediaSurfaceKHR = NULL;
-clEnqueueAcquireDX9MediaSurfacesKHR_fn clEnqueueAcquireDX9MediaSurfacesKHR = NULL;
-clEnqueueReleaseDX9MediaSurfacesKHR_fn clEnqueueReleaseDX9MediaSurfacesKHR = NULL;
 
 static void __OpenCLinitializeD3D9()
 {
     using namespace cv::ocl;
-    static cl_platform_id initializedPlatform = NULL;
-    cl_platform_id platform = (cl_platform_id)Platform::getDefault().ptr();
-    if (initializedPlatform != platform)
+    OpenCLDirectXImpl& impl = getImpl();
+    cl_platform_id platform = impl.getPlatform();
+    if (impl.initializedPlatform9 != platform)
     {
-        clCreateFromDX9MediaSurfaceKHR = (clCreateFromDX9MediaSurfaceKHR_fn)
+        impl.clCreateFromDX9MediaSurfaceKHR = (clCreateFromDX9MediaSurfaceKHR_fn)
```

Comment:
Perhaps such initialization should be moved to `OpenCLDirectXImpl` methods to keep code responsibilities clean.

Reply:
OK, move all the initialization to `OpenCLDirectXImpl`.